sdk/servicebus/service-bus/src/util/atomXmlHelper.ts

```diff
@@ -0,0 +1,356 @@
+  }
+
+  if (webResource.method == "PUT") {
+    webResource.headers.set("content-length", Buffer.byteLength(webResource.body));
+  }
+
+  const response: HttpOperationResponse = await serviceBusAtomManagementClient.sendRequest(
+    webResource
+  );
+
+  try {
+    if (response.bodyAsText) {
+      response.parsedBody = await parseXML(response.bodyAsText, { includeRoot: true });
+    }
+  } catch (err) {
+    const error = new RestError(
+      `ResponseNotInAtomXMLFormat - ${err.message}`,
+      RestError.PARSE_ERROR,
+      response.status,
+      stripRequest(response.request),
+      stripResponse(response)
+    );
+
+    throw error;
+  }
+
+  return serializer.deserialize(response);
+}
+
+/**
+ * Serializes input information to construct the Atom XML request
+ * @param resourceName Name of the resource to be serialized like `QueueDescription`
+ * @param resource The entity details
+ * @param allowedProperties The set of properties that are allowed by the service for the
+ * associated operation(s);
+ */
+export function serializeToAtomXmlRequest(resourceName: string, resource: any): object {
+  const content: any = {};
+  content[resourceName] = resource;
+  content[resourceName][Constants.XML_METADATA_MARKER] = {
+    xmlns: "http://schemas.microsoft.com/netservices/2010/10/servicebus/connect",
+    "xmlns:i": "http://www.w3.org/2001/XMLSchema-instance"
+  };
+
+  content[Constants.XML_METADATA_MARKER] = { type: "application/xml" };
+  const requestDetails: any = {
+    updated: new Date().toISOString(),
+    content: content
+  };
+  requestDetails[Constants.XML_METADATA_MARKER] = {
+    xmlns: "http://www.w3.org/2005/Atom"
+  };
+  return requestDetails;
+}
+
+/**
+ * Transforms response to contain the parsed data.
+ * @param nameProperties The set of 'name' properties to be constructed on the
+ * resultant object e.g., QueueName, TopicName, etc.
+ * @param response
+ * @param shouldParseResponse
+ */
+export async function deserializeAtomXmlResponse(
+  nameProperties: string[],
+  response: HttpOperationResponse
+): Promise<HttpOperationResponse> {
+  let errorBody: any;
+
+  // If received data is a non-valid HTTP response, the body is expected to contain error information
+  if (response.status < 200 || response.status >= 300) {
+    errorBody = response.parsedBody;
+    if (errorBody == undefined) {
+      const HttpResponseCodes: any = Constants.HttpResponseCodes;
+      const statusCode = response.status;
+      if (isKnownResponseCode(statusCode)) {
+        throw buildError(
+          {
+            code: HttpResponseCodes[statusCode]
+          },
+          response
+        );
+      } else {
+        throw buildError(
+          {
+            code: `UnrecognizedHttpResponseStatus: ${statusCode}`
+          },
+          response
+        );
+      }
+    } else {
+      throw buildError(errorBody, response);
+    }
+  }
+
+  response.parsedBody = parseAtomResult(response.parsedBody, nameProperties);
```

Comment:
What happens for a 204 response code, which doesn't have a body?

Reply:
This would result in an empty response body